src/components/side_panel/chart/building_blocks/generic_side_panel/config_panel.ts

```diff
@@ -83,7 +96,40 @@ export class GenericChartConfigPanel extends Component<Props, SpreadsheetChildEn
   }
 
   get dataSetsHaveTitleLabel(): string {
-    return _t("Use row %s as headers", this.calculateHeaderPosition() || "");
+    return this.switchAxes
+      ? _t("Use col %s as headers", numberToLetters(this.calculateHeaderPosition() || 0))
+      : _t("Use row %s as headers", this.calculateHeaderPosition() || "");
+  }
+
+  get switchAxes(): boolean | undefined {
+    return this.props.definition.switchAxes;
+  }
+
+  get canSwitchAxis(): boolean {
+    for (const dataSet of this.dataSets) {
+      try {
+        if (!isXcValid(dataSet.dataRange)) {
```

Comment:
should probably have a dedicated helper that does the try/catch. maybe in isXcValid itself or another with  "isXcRepresentation" ?